packages/timelock-manager/contracts/TimeLockManagerReversible.sol

```diff
@@ -0,0 +1,260 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@api3-contracts/api3-token/contracts/interfaces/IApi3Token.sol";
+import "./interfaces/ITimelockManagerReversible.sol";
+
+/// @title Contract that the API3 DAO uses to timelock API3 tokens
```

Comment:
All right will do

Reply:
I ran the test coverages, I've adjusted the tests for it to pick up stopVesting. Some of the statements that are also present in the original `timelock-manager` are not being covered like:
```
function withdraw()
        external
        override
        onlyIfRecipientHasRemainingTokens(msg.sender)
    {
        address recipient = msg.sender;
        uint256 withdrawable = getWithdrawable(recipient);
        Irequire(withdrawable != 0, "No withdrawable tokens yet");
        timelocks[recipient].remainingAmount = timelocks[recipient]
            .remainingAmount
            .sub(withdrawable);
        require(
            api3Token.transfer(recipient, withdrawable),
            "API3 token transfer failed"
        );
        emit Withdrawn(recipient, withdrawable);
    }
```
After the require statement none of the statements are covered according to istanbul, seems strange because the tests do attempt to withdraw when tokens are available.

Reply:
That's because the coverage package fails if `evm_setNextBlockTimestamp` is used in a test so I was having it skip those tests while measuring coverage:
https://github.com/api3dao/api3-contracts/blob/master/packages/timelock-manager/.solcover.js#L13
The latest version of hardhat / solidity-coverage doesn't have the same limitation though, the dependencies of these packages are way too old.